nameko/testing/pytest.py

```diff
@@ -287,7 +287,9 @@ def socket_creator():
             '127.0.0.1', web_config_port)
         gr = eventlet.spawn(ws_app.run_forever)
         active_sockets.append((gr, ws_app))
-        return wait_for_sock()
+        socket = wait_for_sock()
+        socket.app = ws_app
```

Comment:
I'm happy to delegate all these changes to another PR. I would prefer to land this with the test though.